examples/slicing_ndcube.py

```diff
@@ -0,0 +1,159 @@
+wcs.wcs.cunit = 'arcsec', 'arcsec', 'Angstrom'
+wcs.wcs.cdelt = 10, 10, 0.2
+wcs.wcs.crpix = 2, 2, 0
+wcs.wcs.crval = 1, 1, 10
+wcs.wcs.cname = 'HPC lat', 'HPC lon', 'wavelength'
+# Instantiate the `~ndcube.NDCube`
+example_cube = NDCube(data, wcs=wcs)
+
+##############################################################################
+# So we now have created an `~ndcube.NDCube` named `example_cube`.
+# You may have noticed that the order of the WCS is reversed to the array order - this
+# is normal convention, and something to remember throughout.
+# Now let's first inspect the cube.
+
+##############################################################################
+# Here we can inspect the cube by plotting it
+example_cube.plot()
+
+##############################################################################
+# We can also inspect the dimesions of the cube:
+example_cube.dimensions
+
+##############################################################################
+# We can also inspect the world coordinates for all array elements:
+example_cube.axis_world_coords()
+
+
+##############################################################################
+# Slicing and cropping the cube
+# -----------------------------
+# An `~ndcube.NDCube` can be sliced and cropped both by array indexing (similar to the way a numpy
+# array in indexed) or by real world coordinates. When we use array indices we say we
+# are "slicing" the cube.  When we use world coordinates we say we are "cropping" the cube.
+# Let's begin by slicing by array index.
+
+##############################################################################
+# Slicing by array index
+# ----------------------
+# To slice the `example_cube` so that we extract only one wavelength, we can do:
+# by indexing as such
+sliced_cube = example_cube[1, :, :]
+# here we can see we are left with a 2-D cube which is an image at one wavelength.
+sliced_cube.dimensions
+
+# We can also index a region of interest of the cube at a particular wavelength.
+# Again note that we are slicing here based on the ``array`` index rather than cropping by
+# real world value
+
+sliced_cube = example_cube[1, 10:20, 20:40]
+sliced_cube.dimensions
+
+# now we can inspect the sliced cube, and see it's now a smaller region of interest.
+sliced_cube.plot()
+
+##############################################################################
+# Cropping cube using world coordinate values using `ndcube.NDCube.crop()`
+# ------------------------------------------------------------------------
+# In many cases it's more useful to crop a cube to a region of interest based
+# on real world coordinates such as points in space or over some spectral
+# range. This is achieved by the `ndcube.NDCube.crop()` method which takes high-level astropy coordinate objects,
+# such as `astropy.coordinate.SkyCoord`. The `.crop()` method returns the smallest cube in array-index space that contains all
+# the passed points.
```

Comment:
There's also a crop method on NDCubeSequence so you have to be explicit when to write it as a hyperlink to its documentation.
```suggestion
# such as `astropy.coordinate.SkyCoord`. `ndcube.NDCube.crop()` returns the smallest cube
# in array-index space that contains all the passed points.
```